module/VuFind/tests/unit-tests/src/VuFindTest/Solr/WriterTest.php

```diff
@@ -145,63 +120,64 @@ public function testDeleteAll()
      */
     public function testDeleteRecords()
     {
-        $bm = $this->getBackendManagerWithMockSolr();
-        $connector = $bm->get('Solr')->getConnector();
-        $callback = function ($i): bool {
-            return trim($i->getContent()) == "<?xml version=\"1.0\"?>\n<delete><id>foo</id><id>bar</id></delete>";
-        };
-        $connector->expects($this->once())->method('write')->with($this->callback($callback));
-        $ct = $this->getMockChangeTracker();
-        $ct->expects($this->exactly(2))->method('markDeleted')
-            ->withConsecutive(['biblio', 'foo'], ['biblio', 'bar']);
-        $writer = new Writer($this->getSearchService($bm), $ct);
-        $writer->deleteRecords('Solr', ['foo', 'bar']);
+        $deleteDoc = new DeleteDocument();
+        $deleteDoc->addKeys(['foo', 'bar']);
+        $command = new WriteDocumentCommand('Solr', $deleteDoc);
+        $this->getWriter($command, ['core' => 'biblio'])
+            ->deleteRecords('Solr', ['foo', 'bar']);
     }
 
     /**
-     * Get mock backend manager
+     * Get mock change tracker
      *
-     * @return BackendManager
+     * @return ChangeTracker
      */
-    protected function getBackendManagerWithMockSolr()
+    protected function getMockChangeTracker()
     {
-        $sm = new \Laminas\ServiceManager\ServiceManager();
-        $pm = new BackendManager($sm);
-        $mockBackend
-            = $this->getMockBuilder(\VuFindSearch\Backend\Solr\Backend::class)
-                ->disableOriginalConstructor()
-                ->onlyMethods(['getConnector', 'getIdentifier'])
-                ->getMock();
-        $handlerMap = new HandlerMap();
-        $mockConnector
-            = $this->getMockBuilder(\VuFindSearch\Backend\Solr\Connector::class)
-                ->setConstructorArgs(
-                    [
-                        'http://localhost:8983/solr/biblio',
-                        $handlerMap,
-                        null
-                    ]
-                )
-                ->onlyMethods(['write'])
-                ->getMock();
-        $mockBackend->expects($this->any())->method('getConnector')
-            ->will($this->returnValue($mockConnector));
-        $mockBackend->expects($this->any())->method('getIdentifier')
-            ->will($this->returnValue('Solr'));
-        $sm->setService('Solr', $mockBackend);
-        return $pm;
+        return $this->getMockBuilder(\VuFind\Db\Table\ChangeTracker::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['markDeleted'])
```

Comment:
I don't think this onlyMethods serves a purpose anymore, but I could be wrong.

Reply:
Right, removed.